base_custom_filter/models/ir_filters.py

```diff
@@ -29,3 +30,19 @@ def _selection_type(self):
         ondelete="cascade",
     )
     group_id = fields.Many2one(comodel_name="ir.filters.group", string="Filter Group")
+
+    @api.model
+    def get_filters(self, model, action_id=None):
+        """We need to inject a context to obtain only the records of favorite type."""
+        _self = self.with_context(filter_type="favorite")
+        return super(IrFilters, _self).get_filters(model, action_id=action_id)
```

Comment:
Remember that there's no need of this double variable as we learnt:

```suggestion
        self = self.with_context(filter_type="favorite")
        return super().get_filters(model, action_id=action_id)
```